x-pack/plugins/ml/server/models/fields_service/fields_service.ts

```diff
@@ -185,7 +185,6 @@ export function fieldsServiceProvider({ asCurrentUser }: IScopedClusterClient) {
     } = await asCurrentUser.search({
       index,
       body,
-      // @ts-expect-error @elastic/elasticsearch Datafeed is missing indices_options
```

Comment:
There should be a few more places where this comment exists. 
Searching for `Datafeed is missing indices_options` will find them.

Reply:
@jgowdyelastic there are about 260 errors in the ML plugin. I'd defer to the Codeowners to take care of them since I'm not familiar with this code as good are you are.

Reply:
There have been some unexpected changes to the ML types supplied in the es client. For example, the `Job` interface now as `analysis_config` set to be optional. This appears to be the cause of all of the ML type errors in this PR.
I'm unsure why or when these have changed, perhaps @delvedor or @philkra would know?

Reply:
Our type tester detected that it's not present all the time, I believe it's missing when using the `xpack.info` endpoint.

Reply:
`xpack.info` does not contain ML jobs?
`analysis_config` is definitely not an optional property in an ML Job, perhaps the type tester is confusing and ML Job with a similar looking object?
Looking at the class here:
https://github.com/elastic/elastic-client-generator/blob/master/specification/xpack/usage/Job.ts
I also see the the `detectors` property is now a `JobStatistics` object, rather than a `Detector[]`
It looks to me that perhaps some usage Job object and the actual ML Job have become merged?

Reply:
@jgowdyelastic I updated the package to the `canary.10` version. Mind taking another stab at it?